mri/reconstructors/utils/extract_sensitivity_maps.py

```diff
@@ -60,44 +63,67 @@ def extract_k_space_center_and_locations(data_values, samples_locations,
         if img_shape is None:
             raise ValueError('target image cartesian image shape must be fill')
         raise NotImplementedError
+    if data_values.ndim > 2:
+        warnings.warn('Data Values seem to have rank '
+                      + str(data_values.ndim) +
+                      ' (>2). Using is_fft for now.')
+        is_fft = True
+    if is_fft:
+        img_shape = np.asarray(data_values[0].shape)
+        mask = convert_locations_to_mask(samples_locations, img_shape)
+        indices = np.where(np.reshape(mask, mask.size))[0]
+        data_ordered = np.asarray([
+            np.reshape(data_values[channel], mask.size)[indices]
+            for channel in range(data_values.shape[0])])
     else:
-        if data_values.ndim > 2:
-            warnings.warn('Data Values seem to have rank '
-                          + str(data_values.ndim) +
-                          ' (>2). Using is_fft for now.')
-            is_fft = True
-        if is_fft:
-            img_shape = np.asarray(data_values[0].shape)
-            mask = convert_locations_to_mask(samples_locations, img_shape)
-            indices = np.where(np.reshape(mask, mask.size))[0]
-            data_ordered = np.asarray([
-                np.reshape(data_values[channel], mask.size)[indices]
-                for channel in range(data_values.shape[0])])
-        else:
-            data_ordered = np.copy(data_values)
+        data_ordered = np.copy(data_values)
+    if window_fun is None:
+        if isinstance(thr, float):
+            thr = (thr,)*samples_locations.shape[1]
         condition = np.logical_and.reduce(
             tuple(np.abs(samples_locations[:, i]) <= thr[i]
                   for i in range(len(thr))))
-        index = np.linspace(0, samples_locations.shape[0]-1,
-                            samples_locations.shape[0], dtype=np.int)
-        index = np.extract(condition, index)
-        center_locations = samples_locations[index, :]
-        data_thresholded = data_ordered[:, index]
-        if density_comp is not None:
-            density_comp = density_comp[index]
-            return data_thresholded, center_locations, density_comp
+    elif isinstance(thr, float):
+        condition = np.sum(np.square(samples_locations), axis=1) <= thr**2
+    else:
+        raise ValueError("threshold type is not supported with select window")
+    index = np.linspace(0, samples_locations.shape[0]-1,
+                        samples_locations.shape[0], dtype=np.int)
+    index = np.extract(condition, index)
+    center_locations = samples_locations[index, :]
+    data_thresholded = data_ordered[:, index]
+    if window_fun is not None:
+        if callable(window_fun):
+            window = window_fun(center_locations)
         else:
-            return data_thresholded, center_locations
+            if window_fun == "Hann" or window_fun == "Hanning":
+                a_0 = 0.5
+            elif window_fun == "Hamming":
+                a_0 = 0.53836
```

Comment:
Could you comment on those 2 values, potentially with a link to a Wikipedia page or sthg